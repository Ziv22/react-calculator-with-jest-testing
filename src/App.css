:root {
  --text-color: orangered;
  --text-color-focus: rgb(0, 30, 95);
}

*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-size: inherit;
}

html {
  width: 100vw;
  height: 100vh;
  font-size: 62.5%;
}

#root,
body,
.app {
  width: 100%;
  height: 100%;
}

.container {
  width: 70%;
  height: 100%;
  margin: auto;

  display: flex;
  padding: 10rem;
  font-size: 2.5rem;
  justify-content: space-around;
}

input {
  width: 20rem;
  height: 5rem;
  border: none;
  border-bottom: 2px solid var(--text-color);
  color: var(--text-color);
}

input:focus {
  border: none;
  outline: none;
  border-bottom: 2px solid var(--text-color-focus);
  caret-color: var(--text-color-focus);
  color: var(--text-color-focus);
}

button,
select {
  height: 5rem;
  width: 10rem;
  border: none;
  background-color: transparent;
  border-radius: 3px;
  text-align: center;
}

button,
select:focus {
  outline: none;
}

button:hover,
select:hover {
}

select {
  border-bottom: 1px solid var(--text-color-focus);
}

button {
  border: 1px solid var(--text-color-focus);
}

.result {
  height: 5rem;
  width: 5rem;
  text-align: center;
  display: inline-block;
  line-height: 2;
}
