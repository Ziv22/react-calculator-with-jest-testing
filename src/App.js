import "./App.css";
import { useState } from "react";
import Results from "./components/Results";
import calculator from "./scripts/calculator";

function App() {
  const [firstNumber, setFirstNumber] = useState("");
  const [secondNumber, setSecondNumber] = useState("");
  const [result, setResult] = useState(0);
  const [operation, setOperation] = useState("add");

  const updateFirstNumber = (event) => {
    setFirstNumber(event.target.value);
  };

  const updateSecondNumber = (event) => {
    setSecondNumber(event.target.value);
  };

  const chooseOperation = (event) => {
    setOperation(event.target.value);
  };

  const calculate = () => {
    operate(calculator[operation]);
  };

  const operate = (operation) => {
    if (operation) {
      const result = operation(+firstNumber, +secondNumber);
      if (result !== undefined || result !== null) {
        setResult(result);
      }
    }
  };

  return (
    <div className="App">
      <input
        type="number"
        name="firstNumber"
        id="firstNumber"
        value={firstNumber}
        onChange={updateFirstNumber}
      />

      <select
        name="operation"
        id="operation"
        selected="add"
        onChange={chooseOperation}
      >
        <option value="add">+</option>
        <option value="subtract">-</option>
        <option value="multiply">*</option>
        <option value="divide">/</option>
      </select>

      <input
        type="number"
        name="secondNumber"
        id="secondNumber"
        value={secondNumber}
        onChange={updateSecondNumber}
      />

      <button onClick={calculate}>=</button>

      <Results result={result} />
    </div>
  );
}

export default App;
