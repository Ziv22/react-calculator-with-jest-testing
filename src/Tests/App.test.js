import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../App";

afterEach(cleanup);

test("App component matches snapshot", () => {
  const appSnapshot = renderer.create(<App />).toJSON();
  expect(appSnapshot).toMatchSnapshot();
});
