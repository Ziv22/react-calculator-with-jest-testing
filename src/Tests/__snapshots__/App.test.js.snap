// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div
  className="App"
  data-testid="App"
>
  <input
    data-testid="firstNumber"
    id="firstNumber"
    name="firstNumber"
    onChange={[Function]}
    type="number"
    value=""
  />
  <select
    data-testid="operation"
    id="operation"
    name="operation"
    onChange={[Function]}
    selected="add"
  >
    <option
      value="add"
    >
      +
    </option>
    <option
      value="subtract"
    >
      -
    </option>
    <option
      value="multiply"
    >
      *
    </option>
    <option
      value="divide"
    >
      /
    </option>
  </select>
  <input
    data-testid="secondNumber"
    id="secondNumber"
    name="secondNumber"
    onChange={[Function]}
    type="number"
    value=""
  />
  <button
    data-testid="equalSign"
    onClick={[Function]}
  >
    =
  </button>
  <div
    className="result"
    data-testid="result"
  >
     
    0
     
  </div>
</div>
`;
